README.md

```diff
@@ -1,20 +1,128 @@
 # raft-rs
 
-A toy implementation to better understand the raft consensus protocol
+A toy implementation to better understand the
+[Raft](https://toidiu.com/reads/In_Search_of_an_Understandable_Consensus_Algorithm_(Extended_Raft).pdf)
+consensus protocol
 
-## High-level
-- replicated state machine
-- durable write ahead log
+## Fig 2: condensensed summary of Raft
 
-- role: leader vs follower
-- struct: state_machine { role,  }
-- enum: operations
+### State
+**Persistent state on all servers:**
+- [ ] "Updated on stable storage before responding to RPCs"
+- [ ] currentTerm latest term server has seen (initialized to 0 on first boot,
```

Comment:
Nit: Would be nice to use `code highlighting` (with backticks) here for variable names, types etc, to make them stand out more easily.
```suggestion
- [ ] `currentTerm` latest term server has seen (initialized to 0 on first boot,
```

Reply:
> Also, would be nice to clean up commit history a bit more

Agreed. We should only be doing rebase merge (github ui will let you do that) but you can only enforce that on the paid version of github from what I can tell.

Reply:
> We should only be doing rebase merge (github ui will let you do that) but you can only enforce that on the paid version of github from what I can tell.

Not sure if that's what you mean, but you can go to [Settings](https://github.com/toidiu/raft-rs/settings) and set
- [ ] Allow merge commits Loading
- [ ] Allow squash merging Loading
- [x] Allow rebase merging